maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/AbstractGeneratorMojo.java

```diff
@@ -204,6 +204,25 @@ public void execute()
                                 + "In the future this error will break the build.\n\n" );
         }
 
+        Set<Artifact> wrongScopedArtifacts = mavenDependenciesNotInProvidedScope();
+        if ( !wrongScopedArtifacts.isEmpty() )
+        {
+            StringBuilder errorMessage = new StringBuilder(
+                "\n\nMaven dependencies of Maven Plugins should be in provided scope.\n"
+                    + "Please make sure that all your dependencies declared in POM having Group Id of\n"
```

Comment:
"having Group Id of" --> whose group ID is

Reply:
fixed